sklearn/metrics/_pairwise_distances_reduction/_base.pyx.tp

```diff
@@ -36,8 +37,9 @@ cdef float64_t[::1] _sqeuclidean_row_norms64_dense(
         intp_t d = X.shape[1]
         float64_t[::1] squared_row_norms = np.empty(n, dtype=np.float64)
 
-    for idx in prange(n, schedule='static', nogil=True, num_threads=num_threads):
-        squared_row_norms[idx] = _dot(d, X_ptr + idx * d, 1, X_ptr + idx * d, 1)
+    with threadpool_limits(limits=1, user_api='blas'):
```

Comment:
I would add a comment here because it's surprising that it's necessary while ``_dot`` is already single threaded. Maybe with a link to the issue

Reply:
> _dot is already single threaded.

Actually it looks like it's not. see https://github.com/scipy/scipy/issues/20294#issuecomment-2015561992. It's weird cause every time I tested it, I only saw a single thread running, no matter the length of the arrays.

Reply:
Maybe we should just use ``nrm2`` as suggested here https://github.com/scipy/scipy/issues/20294#issuecomment-2017871024, even though it implies take a square root and then square back. From an efficiency point of view, it should not have a visible impact since the computation of the norms is not the most costly part of the pairwise distance reduction compuations.

Reply:
I did not know this but `ddot` is not necessarily single-threaded, see https://github.com/scipy/scipy/issues/20294#issuecomment-2015561992